lib/rucio/core/did_meta_plugins/did_meta_plugin_interface.py

```diff
@@ -63,39 +77,38 @@ def set_metadata(
         value: str,
         recursive: bool = False,
         *,
-        session: "Optional[Session]" = None
+        session: "Session | None" = None
     ) -> None:
         """
         Add metadata to data identifier.
 
-        :param scope: The scope name.
+        :param scope: The scope of the DID.
         :param name: The data identifier name.
         :param key: the key.
         :param value: the value.
-        :param did: The data identifier info.
-        :param recursive: Option to propagate the metadata change to content.
+        :param recursive: Instruction to propagate the metadata change recursively to content (False by default).
         :param session: The database session in use.
         """
         pass
 
-    @transactional_session
+    @abstractmethod
     def set_metadata_bulk(
         self,
         scope: "InternalScope",
         name: str,
         meta: dict[str, "Any"],
         recursive: bool = False,
         *,
-        session: "Optional[Session]" = None
+        session: "Session | None" = None
```

Comment:
IMO I would wait until py3.10 is the min version before we do this syntax change, so that we can do it without using quotes.